service/lib/agama/dbus/storage/zfcp_controller.rb

```diff
@@ -0,0 +1,153 @@
+          super(path, logger: logger)
+
+          @zfcp_manager = zfcp_manager
+          @controller = controller
+        end
+
+        # Whether the controller is active
+        #
+        # @return [Boolean]
+        def active
+          controller.active?
+        end
+
+        # zFCP channel id
+        #
+        # @return [String]
+        def channel
+          controller.channel || ""
+        end
+
+        # Available WWPNs
+        #
+        # @return [Array<String>]
+        def find_wwpns
+          zfcp_manager.find_wwpns(controller.channel)
+        end
+
+        # Available LUNs
+        #
+        # @param wwpn [String]
+        # @return [Array<String>]
+        def find_luns(wwpn)
+          zfcp_manager.find_luns(controller.channel, wwpn)
+        end
+
+        # Activates the controller
+        #
+        # @note: If "allow_lun_scan" is active, then all LUNs are automatically activated.
+        #
+        # @return [Integer] Exit code of the chzdev command
+        def activate
+          zfcp_manager.activate_controller(controller.channel)
+        end
+
+        # Activates a zFCP disk
+        #
+        # @param wwpn [String]
+        # @param lun [String]
+        #
+        # @return [Integer] Exit code of the chzdev command
+        def activate_disk(wwpn, lun)
+          zfcp_manager.activate_disk(controller.channel, wwpn, lun)
+        end
+
+        # Deactivates a zFCP disk
+        #
+        # @note: If "allow_lun_scan" is active, then the disk cannot be deactivated.
+        #
+        # @param wwpn [String]
+        # @param lun [String]
+        #
+        # @return [Integer] Exit code of the chzdev command
+        def deactivate_disk(wwpn, lun)
+          zfcp_manager.deactivate_disk(controller.channel, wwpn, lun)
+        end
+
+        # Sets the represented zFCP controller
+        #
+        # @note A properties changed signal is emitted if the controller changes.
+        #
+        # @param value [Agama::Storage::ZFCP::Controller]
+        def controller=(value)
+          emit_signal = controller != value
+
+          @controller = value
+          return unless emit_signal
+
+          properties = interfaces_and_properties[ZFCP_CONTROLLER_INTERFACE]
+          dbus_properties_changed(ZFCP_CONTROLLER_INTERFACE, properties, [])
+        end
+
+        ZFCP_CONTROLLER_INTERFACE = "org.opensuse.Agama.Storage1.ZFCP.Controller"
+        private_constant :ZFCP_CONTROLLER_INTERFACE
+
+        # This interface does not have a method to deactivate a controller (controller deactivation
+        # could be problematic). For more details see https://github.com/openSUSE/agama/pull/594.
+        dbus_interface ZFCP_CONTROLLER_INTERFACE do
+          dbus_reader(:active, "b")
+
+          dbus_reader(:channel, "s")
+
+          dbus_method(:GetWWPNs, "out result:as") { [find_wwpns] }
+
+          dbus_method(:GetLUNs, "in wwpn:s, out result:as") { |wwpn| [find_luns(wwpn)] }
```

Comment:
https://en.wikipedia.org/wiki/Logical_unit_number ?
example is especially important because I know LUNs as `[4:0:0:0]` and here it seems to be `0x0005000000000000`